packages/gatsby-source-nacelle/gatsby-node.js

```diff
@@ -0,0 +1,212 @@
+const fs = require('fs-extra');
+const { print } = require('gatsby/graphql');
+const { createRemoteFileNode } = require('gatsby-source-filesystem')
+const {
+  sourceAllNodes,
+  createSchemaCustomization,
+  compileNodeQueries,
+  readOrGenerateDefaultFragments,
+  buildNodeDefinitions,
+  loadSchema,
+  createDefaultQueryExecutor
+} = require('gatsby-graphql-source-toolkit');
+require('dotenv').config();
+
+const CHUNK_SIZE = 100;
+const fragmentsDir = process.cwd() + `/nacelle-fragments`;
+
+const PaginateNacelle = {
+  name: 'NacellePagination',
+  expectedVariableNames: ['first', 'after'],
+  start() {
+    return {
+      variables: { first: CHUNK_SIZE, after: undefined },
+      hasNextPage: true
+    };
+  },
+  next(state, page) {
+    return {
+      variables: { first: CHUNK_SIZE, after: page.nextToken },
+      hasNextPage: Boolean(page.nextToken && page.items.length === CHUNK_SIZE)
+    };
+  },
+  concat(acc, page) {
+    return {
+      ...acc,
+      items: {
+        ...acc.items,
+        ...page.items
+      },
+      nextToken: page.nextToken
+    };
+  },
+  getItems(pageOrResult) {
+    return pageOrResult.items;
+  }
+};
+
+async function createSourcingConfig(gatsbyApi, pluginOptions) {
+  const { verbose } = pluginOptions
+  const nacelleSpaceId =
```

Comment:
While I do appreciate the flexibility here, let's stick to a standard no? I think we should push the camelCase option and would rather throw a meaningful error if it's not there than try and be infinitely flexible.

Reply:
I hear that. I'll add in some error handling and remove the `snek_case`.